src/documents/management/commands/document_consumer.py

```diff
@@ -40,14 +41,34 @@ def _tags_from_path(filepath) -> set[Tag]:
     """
     tag_ids = set()
     path_parts = Path(filepath).relative_to(settings.CONSUMPTION_DIR).parent.parts
-    for part in path_parts:
+    for index, part in enumerate(path_parts):
+        # If first subdir should be interpreted as owner
+        # this subdir should not be added as a tag
+        if index == 0 and settings.CONSUMER_SUBDIR_AS_OWNER:
+            owner_id = User.objects.get(username__iexact=part).pk
+            if owner_id:
+                continue
         tag_ids.add(
             Tag.objects.get_or_create(name__iexact=part, defaults={"name": part})[0].pk,
         )
 
     return tag_ids
 
 
+def _owner_from_path(filepath) -> int:
```

Comment:
In general, I feel like this and `_tags_from_path` could be combined.  There's duplication in particular the initial path parsing.  Returning a dataclass of tuple with the owner id (or none) and the list of tag ids (or none), depending on settings and matching